src/test-utils/render.tsx

```diff
@@ -31,23 +31,34 @@ const WithProviders = ({ children }: { children?: ReactNode }) => {
       <Provider store={store}>
         <QueryClientProvider client={queryClient}>
           <HashRouter>
-            <Web3Provider>
-              <MockedProvider>
-                <BlockNumberProvider>
-                  <ThemeProvider>{children}</ThemeProvider>
-                </BlockNumberProvider>
-              </MockedProvider>
-            </Web3Provider>
+            {/*
+             * Web3Provider is mocked through setupTests.ts
+             * To test behavior that depends on Web3Provider, use jest.unmock('@web3-react/core')
+             */}
+            <MockedProvider>
+              <BlockNumberProvider>
+                <ThemeProvider>{children}</ThemeProvider>
+              </BlockNumberProvider>
+            </MockedProvider>
           </HashRouter>
         </QueryClientProvider>
       </Provider>
     </MockedI18nProvider>
   )
 }
 
-const customRender = (ui: ReactElement) => render(ui, { wrapper: WithProviders })
-const customRenderHook = <Result, Props>(hook: (initialProps: Props) => Result) =>
-  renderHook(hook, { wrapper: WithProviders })
+type CustomRenderOptions = Omit<RenderOptions, 'wrapper'>
+const customRender = (ui: ReactElement, options?: CustomRenderOptions) => {
```

Comment:
Correct me if I am wrong, but I didn't see you pass 2nd argument to `render` anywhere in this PR. Is this something you plan to use in the future?

Reply:
I added this here for completeness with the wrapped library. It is currently unused.